### PR TITLE
Add keyboard selection

### LINKED ISSUE #1910: MenuItem keyboard selection
We currently don't have a clear example on how `MenuItem` is selected/triggered when pressing Enter key as it also triggers the `onClick` (due to event bubbling)  and causes some unwanted behaviours.

Notes: 
* Should we expect users to add key handlers on `MenuItem` like they do with `onClick`? 
  * Maybe we just need to add a shim for event bubbling for keystrokes (like we have for `MenuItem.onClick`)
    * This might break anchor element focus on close?  
* https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-links.html

### PR DESCRIPTION
**Pull request by laliq**
Resolves #1910

### CODE REVIEW

**Review by oddvernes — CHANGES_REQUESTED** [review]
Currently the keypress only handles onclose, which only closes the menu without selecting anything. But it is onClick we need. This is what is used to call different functions. The problem is onClick expects a mouseevent . We need enter and click to behave exactly the same.
My suggestion was to just dispatch a click event when you press enter, which does seem to work perfectly
`      if (key === 'Enter') {
        e.target.dispatchEvent(new Event('click', { bubbles: true }))
      }`
But maybe @mimarz have some some reservations?:P

Also in menu stories something is trapping focus on the anchor button while tabbing, but this seems to be an existing bug, having to do with event.preventDefaut/stopPropagation in the stories maybe

File: packages/eds-core-react/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -101,6 +101,8 @@ export type MenuItemProps = {
   disabled?: boolean
   /** onClick handler */
   onClick?: (e: React.MouseEvent) => void
+  /** onClose handler */
+  onClose?: (e: React.KeyboardEvent) => void
```
Comment (mimarz) [inline packages/eds-core-react/src/components/Menu/MenuItem.tsx:105]: Suggest removing this as we already have an `onClose` passed on top level and it deviates from intended expectation with click vs. keydown.

**Review by mimarz — CHANGES_REQUESTED** [review]
We need to make sure the expected behaviour is the same for when either click or keypress is used. That being that the `onClose` handler passed to `Menu` is called with the event. 

Introducing an `onClose` on `MenuItem` in addition to the on one `Menu` would cause confusion as on or the other would be called be it click or keypress.

**Comment by mimarz** [comment]
> Currently the keypress only handles onclose, which only closes the menu without selecting anything. But it is onClick we need. This is what is used to call different functions. The problem is onClick expects a mouseevent . We need enter and click to behave exactly the same. My suggestion was to just dispatch a click event when you press enter, which does seem to work perfectly ` if (key === 'Enter') { e.target.dispatchEvent(new Event('click', { bubbles: true })) }` But maybe @mimarz have some some reservations?:P
> 
> Also in menu stories something is trapping focus on the anchor button while tabbing, but this seems to be an existing bug, having to do with event.preventDefaut/stopPropagation in the stories maybe

Not sure, mimicing a click event when a key is pressed feels very wrong :P Typing should not be a problem as we can just use "or" on the event type?

**Comment by oddvernes** [comment]
> Not sure, mimicing a click event when a key is pressed feels very wrong :P Typing should not be a problem as we can just use "or" on the event type?

`onClick?: (e: React.MouseEvent | React.KeyboardEvent) => void `

`      if (key === 'Enter') {
        if (onClick) {
          onClick(e)
        }`

Yes this works if you also declare the event as mouse or keyboard downstream in the stories (I don't know how many people even use this event in their apps, so its only a small breaking change for those that do)

**Comment by mimarz** [comment]
> 

I was thinking more for the `onClose` handler? Lets not mess expected native behaviours on handler 😅

**Comment by laliq** [comment]
Thanks guys 🙏

**Comment by laliq** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onclick

**Review by oddvernes — APPROVED** [review]
LGTM 👍 
I'll make a new issue for focus getting stuck on the anchor button in menu stories

**Review by mimarz — APPROVED** [review]
Quirky solution! LGTM 👍

**Pull request merged**